### ISSUE TITLE
monix as a maven dependency fails

### ISSUE THREAD

**Issue by bbarker** [monix/monix#332]
I apologize in advance if this is unrelated to monix, but I've had the issue on two systems, and only for the monix dependency, not other library dependencies.

When using it from maven:

```xml
        <dependency>
            <groupId>io.monix</groupId>
            <artifactId>monix_${scala.version.major}</artifactId>
            <version>2.2.1</version>
            <scope>provided
```

I always get the following error:

```
[ERROR] Failed to execute goal on project ced2ar3-jsgen: Could not resolve dependencies for project edu.ncrn.cornell:ced2ar3-jsgen:jar:0.0.1-SNAPSHOT: Failure to find io.monix:monix_2.12:jar:2.2.1 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

I've also tried it with a few different versions (2.2.1, 2.2.2, 2.2.3) of monix and have the same problem.

**Comment by alexandru** [comment]
@bbarker thanks for the report, please tell me the Maven version that you're using.

Monix is a modular project. What we did for the main Monix package (`"io.monix" %% "monix"`) is to apply `<packaging>pom</packaging>`, which specifies that this is a meta project that has sub-projects as dependencies, so the POM file itself is the delivered artifact, instead of a JAR file. Therefore its related JAR does not exist.

Resources: 

- The PR that made this change: https://github.com/monix/monix/pull/298
- Maven's documentation on project aggregation: https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Project_Aggregation
- This problem discovered and fixed in Coursier: https://github.com/coursier/coursier/issues/429

In the meantime, you can always depend on `monix-reactive` instead of `monix` and the problem should go away.

**Comment by alexandru** [comment]
Also, please use latest `2.2.3` since we have had bug fixes since `2.2.1`. Thanks.

**Comment by bbarker** [comment]
Thanks! The workaround works for me. I'm using mvn 3.3.9, bundled with IntelliJ. This also prompted me to switch to Coursier for the first time in the scala.js sub module that uses sbt ... 👍
